### PR TITLE
feat(docs): replace npm@<7 with yarn

### PR DESCRIPTION
**Pull request by gpoole**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[ ] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

Update the install command in the readme for peer dependencies to quote strings with characters interpreted by shells like bash and zsh.

**Which issue (if any) does this pull request address?**

The install command as currently given is not usable in zsh without escaping the `^`s.

### CODE REVIEW

**Comment by mightyiam** [comment]
Does anyone still such old npm?

Wouldn't we rather remove this altogether?

**Comment by gpoole** [comment]
Yeah not sure about that version of NPM. The reason I needed it was for Yarn since (unless I've missed something) it doesn't seem to install peer deps automatically either. How about keeping it and changing it to a Yarn command?

**Comment by mightyiam** [comment]
Really? Yarn behaves this way? Well... soon some day we won't have plugins as peerDeps. So, if anything, I would remove this.

**Comment by mightyiam** [comment]
Sorry, I didn't mean to close. Would you like to remove it?

**Comment by gpoole** [comment]
Yes, looks like this is Yarn's chosen behaviour for the present anyway @mightyiam: https://github.com/yarnpkg/yarn/issues/1503#issuecomment-753624157

I do think it's useful for Yarn users right now even if you're planning to remove it later, so would you consider keeping it in Yarn form until the time peer deps aren't required at all? I think removing it entirely will make life harder for Yarn users, since we'd have to figure out the install command for the peer deps from scratch instead of just modifying the handy npm command you have there, so if you'd rather not switch it to a Yarn command I'd say I'd definitely vote for leaving it alone.

**Comment by mightyiam** [comment]
Okay, so we should replace the existing section with a yarn specific one.

**Comment by gpoole** [comment]
Updated with a Yarn sample, does that look ok?

File: readme.md [hunk 1]
```diff
@@ -29,22 +29,22 @@ This can be achieved by:
...
+```sh
+yarn add \
   "typescript@*" \
   "eslint@^8.0.1" \
   "eslint-plugin-promise@^6.0.0" \
   "eslint-plugin-import@^2.25.2" \
   "eslint-plugin-n@^15.0.0" \
   "@typescript-eslint/eslint-plugin@^6.4.0" \
-  eslint-config-standard-with-typescript@latest
+  eslint-config-standard-with-typescript
```
Comment (mightyiam) [inline readme.md]: Is this change related?

File: readme.md [hunk 2]
```diff
@@ -29,22 +29,22 @@ This can be achieved by:
...
-`npm@<7` does not automatically install `peerDependencies`,
+Yarn does not automatically install `peerDependencies`,
 so if that's what you're using, install them manually.
 Here is an example, but use it only for reference,
 because your decisions regarding version ranges and range specifiers may vary.
 
-```
-npm install --save-dev \
+```sh
+yarn add \
```
Comment (mightyiam) [inline readme.md]: Is there no equivalent to `--save-dev`?

File: readme.md [hunk 3]
```diff
@@ -29,22 +29,22 @@ This can be achieved by:
...
 
-`npm@<7` does not automatically install `peerDependencies`,
+Yarn does not automatically install `peerDependencies`,
 so if that's what you're using, install them manually.
 Here is an example, but use it only for reference,
 because your decisions regarding version ranges and range specifiers may vary.
 
-```
-npm install --save-dev \
+```sh
```
Comment (mightyiam) [inline readme.md:39]: Unrelated change.

File: readme.md (hunk 2 above)
Comment (gpoole) [reply readme.md]: There is, I've added it in a second commit after this one.

File: readme.md (hunk 1 above)
Comment (gpoole) [reply readme.md]: If you specify a version explicitly, `yarn add` will put the literal version specified into your `package.json`, unlike npm which will resolve `latest` to a specific patch version range and write that. Leaving the version specifier off will implicitly request `latest`.

File: readme.md (hunk 3 above)
Comment (gpoole) [reply readme.md]: Reverted

**Comment by mightyiam** [comment]
Seems like a test needs to be amended.
